solidity/test/bridge/Bridge.test.ts

```diff
@@ -968,1024 +1023,1171 @@ describe("Bridge", () => {
-          // To pass the proof validation, the difficulty returned by the relay
-          // must be 22350181 for test data used in this scenario. Setting
-          // a different value will cause difficulty comparison failure.
-          data.chainDifficulty = 1
+            before(async () => {
+              await createSnapshot()
 
-          await expect(runSweepScenario(data)).to.be.revertedWith(
-            "Not at current or previous difficulty"
-          )
-        })
-      })
+              // Simulate the wallet is an Live one and is known in
+              // the system.
+              await bridge.setWallet(walletPubKeyHash, {
+                ecdsaWalletID: ethers.constants.HashZero,
+                mainUtxoHash: ethers.constants.HashZero,
+                pendingRedemptionsValue: 0,
+                createdAt: await lastBlockTime(),
+                moveFundsRequestedAt: 0,
+                state: walletState.Live,
+              })
 
-      context("when headers chain length is not valid", () => {
-        const data: SweepTestData = JSON.parse(
-          JSON.stringify(SingleP2SHDeposit)
+              // Necessary to pass the first part of proof validation.
+              await relay.setCurrentEpochDifficulty(data.chainDifficulty)
+              await relay.setPrevEpochDifficulty(data.chainDifficulty)
+
+              // Deploy another bridge which has higher `txProofDifficultyFactor`
+              // than the original bridge. That means it will need 12 confirmations
+              // to deem transaction proof validity. This scenario uses test
+              // data which has only 6 confirmations. That should force the
+              // failure we expect within this scenario.
+              otherBridge = await Bridge.deploy(
+                bank.address,
+                relay.address,
+                treasury.address,
+                ethers.utils.hexZeroPad("0x01", 20),
+                12
+              )
+              await otherBridge.deployed()
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should revert", async () => {
+              await expect(
+                otherBridge.submitSweepProof(
+                  data.sweepTx,
+                  data.sweepProof,
+                  data.mainUtxo
+                )
+              ).to.be.revertedWith(
+                "Insufficient accumulated difficulty in header chain"
+              )
+            })
+          }
         )
-        // Take wallet public key hash from first deposit. All
-        // deposits in same sweep batch should have the same value
-        // of that field.
-        const { walletPubKeyHash } = data.deposits[0].reveal
+      })
+    })
 
-        before(async () => {
-          await createSnapshot()
+    context("when the wallet state is MovingFunds", () => {
+      // The execution of `submitSweepProof` is the same for wallets in
```

Comment:
I'd even remove the specific assertions altogether and leave just one "should succeed" that will make sure the function doesn't revert when called for a `MovingFunds` wallet.

Reply:
I added this in 15a2d96b8d7e39b8204ae9c3e818259a57782b94:
```      
it("should succeed", async () => {
        await expect(
          bridge.submitSweepProof(
            data.sweepTx,
            data.sweepProof,
            data.mainUtxo
          )
        ).not.to.be.reverted
 })
 ```